src/test/java/se/iths/jakartaeelab1/service/PersonServiceTest.java

```diff
@@ -0,0 +1,126 @@
+
+    @BeforeEach
+    void setUp() {
+        testPerson.setId(testId);
+        testPerson.setName("Test Name");
+        testPerson.setAge(25);
+        testPerson.setProfession("Test Profession");
+    }
+
+    @Test
+    void testAddPerson() {
+        when(personRepository.createPerson(any(Person.class))).thenReturn(testPerson);
+
+        Person result = personService.addPerson(testPersonDto);
+
+        assertNotNull(result);
+        assertEquals("Test Name", result.getName());
+        assertEquals(25, result.getAge());
+        assertEquals("Test Profession", result.getProfession());
+
+        verify(personRepository, times(1)).createPerson(any(Person.class));
+    }
+
+    @Test
+    void testAllPersons() {
+        when(personRepository.readAllPerson()).thenReturn(List.of(testPerson));
+
+        var result = personService.allPersons();
+
+        assertNotNull(result);
+        assertEquals(1, result.persons().size());
+        assertEquals("Test Name", result.persons().get(0).name());
+
+        verify(personRepository).readAllPerson();
+    }
+
+    @Test
+    void testOnePerson() {
+        when(personRepository.findPersonById(testId)).thenReturn(testPerson);
+
+        var result = personService.onePerson(testId);
+
+        assertNotNull(result);
+        assertEquals("Test Name", result.name());
+
+        verify(personRepository).findPersonById(testId);
+    }
+
+    @Test
+    void testUpdatePerson() {
+        when(personRepository.findPersonById(testId)).thenReturn(testPerson);
+        when(personRepository.updatePerson(any(Person.class))).thenAnswer(invocation -> invocation.getArgument(0));
+
+        PersonDto updateDto = new PersonDto("Updated Name", 30, "Updated Profession");
+        var result = personService.updatePerson(testId, updateDto);
+
+        assertNotNull(result);
+        assertEquals("Updated Name", result.name());
+        assertEquals(30, result.age());
+        assertEquals("Updated Profession", result.profession());
+
+        verify(personRepository).updatePerson(any(Person.class));
+    }
+
+    @Test
+    void testRemovePerson() {
+        when(personRepository.findPersonById(testId)).thenReturn(testPerson);
+        doNothing().when(personRepository).deletePerson(testId);
+
+        personService.removePerson(testId);
+
+        verify(personRepository).deletePerson(testId);
+    }
+
+    @Test
+    void testOnePersonNotFoundException() {
+        UUID nonExistentId = UUID.randomUUID();
+        when(personRepository.findPersonById(nonExistentId)).thenReturn(null);
+
+        NotFoundException thrownException = assertThrows(
+                NotFoundException.class,
+                () -> personService.onePerson(nonExistentId),
+                "Expected onePerson() to throw, but it didn't"
+        );
+
+        assertTrue(thrownException.getMessage().contains("Invalid id" + nonExistentId));
+        verify(personRepository).findPersonById(nonExistentId);
+    }
```

Comment:
This seems like a long way to do this: 
```suggestion
    @Test
    void testOnePersonNotFoundException() {
       UUID nonExistentId = UUID.randomUUID();

        assertThrows(NotFoundException.class, () -> personService.onePerson(nonExistentId));
        verify(personRepository).findPersonById(nonExistentId);
    }
```
personRepository.findPersonById() will return null by default as it is mocked. 
Is there a reason to declare a variable for NotFoundException? and not just assert that it is thrown by onePerson(nonExistentId)?

Reply:
You're right, changing it now.